sdk/core/azure-core/src/main/java/com/azure/core/util/UserAgentUtil.java

```diff
@@ -0,0 +1,118 @@
+     * application ID will be prepended and platform telemetry will be appended, a fully configured User-Agent header
+     * format is <application_id> azsdk-java-<client_lib>/<sdk_version> <platform_info>.
+     */
+    private static final String DEFAULT_USER_AGENT_FORMAT = DEFAULT_USER_AGENT_HEADER + "-%s/%s";
+
+    // From the design guidelines, the platform info format is:
+    // <language runtime>; <os name> <os version>
+    private static final String PLATFORM_INFO_FORMAT = "%s; %s; %s";
+
+    private UserAgentUtil() {
+        // don't instantiate
+    }
+
+    /**
+     * Return user agent string for the given sdk name and version.
+     *
+     * @param sdkName name of the SDK.
+     * @param sdkVersion Version of the SDK.
+     * @return User agent string as specified in design guidelines.
+     */
+    public static String toUserAgentString(String sdkName, String sdkVersion) {
+        return toUserAgentString(null, sdkName, sdkVersion);
+    }
+
+    /**
+     * Return user agent string for the given sdk name and version.
+     *
+     * @param sdkName Name of the SDK.
+     * @param sdkVersion Version of the SDK.
+     * @param configuration The configuration to use to determine if platform info should be included in the user agent
+     * string.
+     * @return User agent string as specified in design guidelines.
+     */
+    public static String toUserAgentString(String sdkName, String sdkVersion,
+        Configuration configuration) {
+        return toUserAgentString(null, sdkName, sdkVersion, configuration);
+    }
+
+    /**
+     * Return user agent string for the given sdk name and version.
+     *
+     * @param applicationId Name of the application.
+     * @param sdkName Name of the SDK.
+     * @param sdkVersion Version of the SDK.
+     * @return User agent string as specified in design guidelines.
+     */
+    public static String toUserAgentString(String applicationId, String sdkName, String sdkVersion) {
+        return toUserAgentString(applicationId, sdkName, sdkVersion, null);
+    }
+
+    /**
+     * Return user agent string for the given sdk name and version.
+     *
+     * @param applicationId Name of the application.
+     * @param sdkName Name of the SDK.
+     * @param sdkVersion Version of the SDK.
+     * @param configuration The configuration to use to determine if platform info should be included in the user agent
+     * string.
+     * @return User agent string as specified in design guidelines.
+     */
+    public static String toUserAgentString(String applicationId, String sdkName, String sdkVersion,
+        Configuration configuration) {
+        StringBuilder userAgentBuilder = new StringBuilder();
+
+        // Only add the application ID if it is present as it is optional.
+        if (applicationId != null) {
+            applicationId = applicationId.length() > 24 ? applicationId.substring(0, 24) : applicationId;
```

Comment:
Can 24 be made into a private constant? So if we need to change it, we can without missing a place. Also, how did we come up with 24? :)